src/controller/java/src/chip/devicecontroller/model/AttributeWriteRequest.java

```diff
@@ -36,36 +36,37 @@ private AttributeWriteRequest(
     this.endpointId = endpointId;
     this.clusterId = clusterId;
     this.attributeId = attributeId;
-    this.tlv = tlv;
+    this.tlv = tlv.clone();
     this.dataVersion = dataVersion;
   }
 
-  public ChipPathId getEndpointId() {
+  public final ChipPathId getEndpointId() {
```

Comment:
nit: doesn't hurt at all so feel free to ignore, but the `final` modifier is redundant for methods in a `final` class.

Reply:
ok, updated, thanks